contrib/build-linux/sdist/make_sdist.sh

```diff
@@ -6,45 +6,44 @@ PROJECT_ROOT="$(dirname "$(readlink -e "$0")")/../../.."
 CONTRIB="$PROJECT_ROOT/contrib"
 CONTRIB_SDIST="$CONTRIB/build-linux/sdist"
 DISTDIR="$PROJECT_ROOT/dist"
-LOCALE="$PROJECT_ROOT/electrum/locale/"
+LOCALE="$PROJECT_ROOT/electrum/locale"
 
 . "$CONTRIB"/build_tools_util.sh
 
-# note that at least py3.7 is needed, to have https://bugs.python.org/issue30693
-python3 --version || fail "python interpreter not found"
+if ([ "$OMIT_UNCLEAN_FILES" != 1 ]); then
+  # note that at least py3.7 is needed, to have https://bugs.python.org/issue30693
+  python3 --version || fail "python interpreter not found"
 
-break_legacy_easy_install
+  break_legacy_easy_install
 
-# upgrade to modern pip so that it knows the flags we need.
-# we will then install a pinned version of pip as part of requirements-build-sdist
-python3 -m pip install --upgrade pip
+  # upgrade to modern pip so that it knows the flags we need.
+  # we will then install a pinned version of pip as part of requirements-build-sdist
+  python3 -m pip install --upgrade pip
 
-info "Installing pinned requirements."
-python3 -m pip install --no-dependencies --no-warn-script-location -r "$CONTRIB"/deterministic-build/requirements-build-sdist.txt
+  info "Installing pinned requirements."
+  python3 -m pip install --no-dependencies --no-warn-script-location -r "$CONTRIB"/deterministic-build/requirements-build-sdist.txt
 
 
-"$CONTRIB"/make_packages || fail "make_packages failed"
+  "$CONTRIB"/make_packages || fail "make_packages failed"
+fi
```

Comment:
What's going on here, why is this whole block guarded by that envvar? Presumably you only want to guard the `make_packages` call?

If `pip install deterministic-build/requirements-build-sdist.txt` is not run, is the build even reproducible?
On line53, `python3 setup.py sdist` is run, and the generated tar.gz might very well depend on the setuptools version used.